core/trino-main/src/main/java/io/trino/server/QueryResource.java

```diff
@@ -71,20 +72,19 @@ public QueryResource(DispatchManager dispatchManager, AccessControl accessContro
 
     @ResourceSecurity(AUTHENTICATED_USER)
     @GET
-    public List<BasicQueryInfo> getAllQueryInfo(@QueryParam("state") String stateFilter, @Context HttpServletRequest servletRequest, @Context HttpHeaders httpHeaders)
+    public List<BasicQueryInfo> getAllQueryInfo(@QueryParam("state") Set<String> stateFilters, @Context HttpServletRequest servletRequest, @Context HttpHeaders httpHeaders)
     {
-        QueryState expectedState = stateFilter == null ? null : QueryState.valueOf(stateFilter.toUpperCase(Locale.ENGLISH));
+        Set<QueryState> expectedStates = stateFilters.stream()
+                .map(f -> f.toUpperCase(Locale.ENGLISH))
+                .map(QueryState::valueOf)
+                .collect(Collectors.toUnmodifiableSet());
 
         List<BasicQueryInfo> queries = dispatchManager.getQueries();
         queries = filterQueries(sessionContextFactory.extractAuthorizedIdentity(servletRequest, httpHeaders), queries, accessControl);
 
-        ImmutableList.Builder<BasicQueryInfo> builder = ImmutableList.builder();
-        for (BasicQueryInfo queryInfo : queries) {
-            if (stateFilter == null || queryInfo.getState() == expectedState) {
-                builder.add(queryInfo);
-            }
-        }
-        return builder.build();
+        return queries.stream()
+                .filter(q -> expectedStates.isEmpty() || expectedStates.contains(q.getState()))
```

Comment:
the expectedStates.isEmpty is not needed in that form.

Return early instead:

```
if (expectedStates.isEmpty()) {
    return queries;
}
```

Reply:
yep, better.